libvirt/resource_libvirt_domain.go

```diff
@@ -344,16 +346,31 @@ func resourceLibvirtDomainCreate(d *schema.ResourceData, meta interface{}) error
 			if err != nil {
 				return fmt.Errorf("Error retrieving network name: %s", err)
 			}
+			dhcpMutex.Lock()
 			networkDef, err := newDefNetworkfromLibvirt(&network)
+			dhcpMutex.Unlock()
```

Comment:
Oh, I see, I think it is for avoiding concurrent changes made by `enableDHCP`, right?